com.unity.cinemachine/Editor/Upgrader/CinemachineUpgradeManager.cs

```diff
@@ -400,59 +290,98 @@ void UpgradePrefabsAndPrefabInstances()
                             continue; // is a backup copy
 
                         UpgradeObjectComponents(c.gameObject, timelineManager);
+                    }
+
+                    // clean-up now, because we needed obsolete components to restore references
```

Comment:
I'm confused by this.  When did we restore references?

Reply:
In UpgradeObjectComponents, in the foreach above this foreach